configure.ac

```diff
@@ -1110,18 +1110,20 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
   ]], [[
     getauxval(AT_HWCAP);
   ]])],
- [ AC_MSG_RESULT(yes); HAVE_STRONG_GETAUXVAL=1 ],
+ [ AC_MSG_RESULT(yes); HAVE_STRONG_GETAUXVAL=1; AC_DEFINE(HAVE_STRONG_GETAUXVAL, 1, [Define this symbol to build code that uses getauxval)]) ],
  [ AC_MSG_RESULT(no); HAVE_STRONG_GETAUXVAL=0 ]
 )
 
 AC_MSG_CHECKING(for weak getauxval support in the compiler)
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+    #ifdef __linux__
     unsigned long getauxval(unsigned long type) __attribute__((weak));
     #define AT_HWCAP 16
```

Comment:
This seems very fragile. On what target is getauxval *only* weak?

Reply:
Why would it be (more) fragile? Weak linking is already pretty fragile AFAIK.
IMO this weak linking passes always. See masters [Win64 build](https://bitcoinbuilds.org/index.php?ansilog=0c93cb58-e6da-4394-b703-6399079e7bef.log#l593) as an example.

AFAIK this check is part of googles crc32c suite.
Limiting it to linux makes absolute sense to me since `getauxval` (it doesn't exists on Mac/win).

Reply:
Sorry, I do mean the weak linking entirely. What do we lose, if we *only* check for strong linking?

Reply:
That's a valid question.
AFAIK the google crc32c `getauxval` weak linking check is for android (< API level 20).
Crc32c is an upstream upstream subtree. I guess if we don't want to check for the weak linking, we might want to PR that there.

Reply:
I assume if we just don't define it, that code block simply won't be used?

Reply:
> This seems very fragile. On what target is getauxval only weak?

a) getauxval only exists on Linux
b) by nature of weak linking, it *could* weakly link on every platform, this makes no sense though